XForm/XForm/Functions/String/Replace.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Microsoft.CodeAnalysis.Elfie.Model.Strings;
+using System;
+using XForm.Commands;
+using XForm.Data;
+using XForm.Types;
+
+namespace XForm.Functions
+{
+    internal class ReplaceBuilder : IFunctionBuilder
+    {
+        public string Name => "Replace";
+        public string Usage => "Replace([InputColumnName], [FindText], [ReplaceText?], [ExactMatch?])";
+
+        public IDataBatchColumn Build(IDataBatchEnumerator source, WorkflowContext context)
+        {
+            return ReplaceColumn.Build(
+                source,
+                context.Parser.NextColumn(source, context),
+                context.Parser.NextString(),
+                context.Parser.NextString(),
+                (context.Parser.HasAnotherArgument ? context.Parser.NextBoolean() : false));
+        }
+    }
+
+    public class ReplaceColumn : IDataBatchColumn
+    {
+        private IDataBatchColumn Column { get; }
+        private String8 FindText { get; }
+        private String8 ReplaceText { get; }
+        private bool ExactMatch { get; }
+
+        private ReplaceColumn(IDataBatchColumn column, String8 findText, String8 replaceText, bool exactMatch)
+        {
+            Column = column;
+            FindText = findText;
+            ReplaceText = replaceText;
+            ExactMatch = exactMatch;
+        }
+
+        public static IDataBatchColumn Build(IDataBatchEnumerator source, IDataBatchColumn inputColumn, string findText, string replaceText, bool exactMatch)
+        {
+            // Replace only works on String8 columns
+            if (inputColumn.ColumnDetails.Type != typeof(String8))
+            {
+                return inputColumn;
```

Comment:
I think we should throw ArgumentException here instead of quietly doing nothing.